BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -87,4 +87,41 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
     return self;
 }
 
+- (NSDictionary *)composeJSONData
+{
+    NSMutableDictionary *JSONData = [NSMutableDictionary new];
+    if (self.type) {
+        JSONData[BOXAPIObjectKeyRepresentation] = self.type;
+    }
+    if (self.dimensions) {
+        JSONData[BOXAPIObjectKeyProperties] = @{BOXAPIObjectKeyDimensions: self.dimensions};
+    }
+    if (self.status || self.statusCode) {
+
+        NSMutableDictionary *statusDict = [NSMutableDictionary new];
+        if (self.status) {
+            statusDict[BOXAPIObjectKeyState] = self.status;
+        }
+        if (self.statusCode) {
+            statusDict[BOXAPIObjectKeyCode] = self.statusCode;
+        }
+        JSONData[BOXAPIObjectKeyStatus] = statusDict;
+    }
+    if (self.details) {
+        JSONData[BOXAPIObjectKeyDetails] = self.details;
+    }
+
+    NSString *contentURLString = self.contentURL.path;
+    if (contentURLString && [self.type isEqualToString:BOXRepresentationTypeHLS]) {
+        contentURLString = [contentURLString stringByReplacingOccurrencesOfString:BOXRepresentationTemplateValueHLSManifest withString:BOXRepresentationTemplateKeyAccessPath];
```

Comment:
Haha. Maybe we should just store the template in the first place, as well?

Reply:
want to preserve current behavior in this scope

Reply:
I contend that it wouldn't change any observable behavior.

Reply:
updated